pkg/rollout/rollout.go

```diff
@@ -148,17 +148,18 @@ func (r *Rollout) UpdateService(svc *run.Service) (*run.Service, error) {
 		return nil, errors.Wrapf(err, "failed to diagnose health for candidate %q", candidate)
 	}
 
-	svc, err = r.updateServiceBasedOnDiagnosis(svc, diagnosis.OverallResult, stable, candidate)
+	traffic, err := r.trafficBasedOnDiagnosis(svc, diagnosis.OverallResult, stable, candidate)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to update service after diagnosis")
+		return nil, errors.Wrap(err, "failed to configure traffic after diagnosis")
 	}
-	if svc == nil {
-		// If service was unchanged, nil is returned.
-		// TODO(gvso): This should go away once we start getting traffic config
-		// object from updateServiceBasedOnDiagnosis.
```

Comment:
I left it as nil instead of returning the same, unchanged traffic. This is because we need to differentiate between an unchanged traffic because of inconclusive/no enough time since last rollout. So, it is easy to check for that if we get a nil traffic.

We could also use go-cmp to compare the old traffic to the new one to determine if the traffic was unchanged if using nil doesn't sound like a good idea.

Reply:
go-cmp is a bad idea.
However returning `(Traffic, bool)` to indicate a change was made or not is more go-idiomatic.

Reply:
That said you can do that in a separate PR.

Reply:
Should this function return 3 values when that's implemented `(Traffic, bool, error)`?

Reply:
I wouldn't mind, but if traffic==nil implies bool=false, then it's an overkill to have bool.

Reply:
If we add a boolean, we can just return the old traffic with no changes, so we will never have to check `if traffic = nil` since we guarantee a traffic object regardless of the diagnosis/situation.